pkg/aws/services/vpclattice.go

```diff
@@ -20,6 +22,29 @@ type ServiceNetworkInfo struct {
 	SvcNetwork vpclattice.ServiceNetworkSummary
 	Tags       Tags
 }
+
+type LatticeNameProvider interface {
+	LatticeName() string
+}
+
+type NotFoundError struct {
+	ResourceType string
+	Name         string
+}
```

Comment:
Getting something that looked and behaved sensibly for error handling and identifying errors of this type took me a while. For folks more familiar with Go please feel free to suggest an alternate approach if you think there's a better option.